request_handler.cpp

```diff
@@ -66,12 +67,15 @@ void request_handler::handle_request(const request& req, reply& rep)
     rep = reply::stock_reply(reply::not_found);
     return;
   }
+  */
+  //always extension txt
+  std::string extension = "html";
 
   // Fill out the reply to be sent to the client.
   rep.status = reply::ok;
-  char buf[512];
-  while (is.read(buf, sizeof(buf)).gcount() > 0)
-    rep.content.append(buf, is.gcount());
+  for (int i =0; buffer_[i] != '\0' || i < 8192; i++)
```

Comment:
Should this be an &&? Seems like both conditions are required, not one or the other.